plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/type/TypeInfo.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.bigquery.type;
+
+public abstract sealed class TypeInfo
+        permits PrimitiveTypeInfo, ArrayTypeInfo, StructTypeInfo
+{
+    protected TypeInfo() {}
+
+    public abstract Category getCategory();
```

Comment:
We don't need a category - this information is already encoded in the sealed class hierarchy. Instead of a `switch (typeInfo.getCategory())` we can `switch (typeInfo)`